packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -797,12 +795,6 @@ const ComboBox = forwardRef(
       downshiftSetInputValue,
       toggleMenu,
     ]);
-    useEffect(() => {
-      if (inputRef.current) {
-        inputRef.current.setSelectionRange(cursorPosition, cursorPosition);
-      }
-    }, [inputValue, cursorPosition]);
-
```

Comment:
Any idea what this was used for originally? It's wild that this whole useEffect was in there that not only wasn't needed, but actively caused a bug!

Reply:
the cursorPosition came from the recent Autocomplete with Typeahead PR, but the odd cursor behavior was happening before that. When trying to fix the bug though, these additions seemed to probably help a weird cursor position when adding the autocomplete, so I realized fixing the original cursor bug also meant we could get rid of these new additions